internal/util/diff/diff_test.go

```diff
@@ -19,233 +19,226 @@ import (
 	"bufio"
 	"bytes"
 	"io"
-	"io/ioutil"
-	"os"
-	"path/filepath"
 	"strings"
 	"testing"
 
-	"github.com/GoogleContainerTools/kpt/internal/pkg"
 	"github.com/GoogleContainerTools/kpt/internal/testutil"
 	. "github.com/GoogleContainerTools/kpt/internal/util/diff"
-	"github.com/GoogleContainerTools/kpt/internal/util/fetch"
-	kptfilev1alpha2 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1alpha2"
-	"github.com/GoogleContainerTools/kpt/pkg/kptfile/kptfileutil"
 	"github.com/stretchr/testify/assert"
-	"sigs.k8s.io/kustomize/kyaml/copyutil"
 )
 
-// TestCommand_RunRemoteDiff verifies Command can show changes for remote diff
-// operation.
-//
-// 1. add data to the master branch
-// 2. commit and tag the master branch
-// 3. add more data to the master branch, commit it
-// 4. clone at the tag
-// 5. add more data to the master branch, commit it
-// 5. Run remote diff between master and cloned
-func TestCommand_RunRemoteDiff(t *testing.T) {
-	t.SkipNow()
-	g, w, clean := testutil.SetupRepoAndWorkspace(t, testutil.Content{
-		Data:   testutil.Dataset1,
-		Branch: "master",
-	})
-	defer clean()
-
-	// create a commit with dataset2 and tag it v2, then add another commit on top with dataset3
-	commit0, err := g.GetCommit()
-	assert.NoError(t, err)
-	err = g.ReplaceData(testutil.Dataset2)
-	assert.NoError(t, err)
-	_, err = g.Commit("new-data for v2")
-	assert.NoError(t, err)
-	commit, err := g.GetCommit()
-	assert.NoError(t, err)
-	err = g.Tag("v2")
-	assert.NoError(t, err)
-	err = g.ReplaceData(testutil.Dataset3)
-	assert.NoError(t, err)
-	_, err = g.Commit("new-data post-v2")
-	assert.NoError(t, err)
-	commit2, err := g.GetCommit()
-	assert.NoError(t, err)
-	assert.NotEqual(t, commit, commit0)
-	assert.NotEqual(t, commit, commit2)
-
-	localPkg := createPackage(t, g, w)
-
-	diffOutput := &bytes.Buffer{}
-
-	err = (&Command{
-		Path:         localPkg.UniquePath.String(),
-		Ref:          "master",
-		DiffType:     "remote",
-		DiffTool:     "diff",
-		DiffToolOpts: "-r -i -w",
-		Output:       diffOutput,
-	}).Run()
-	assert.NoError(t, err)
+func TestCommand_Diff(t *testing.T) {
```

Comment:
Can these tests follow update tests style with updated upstream, initial upstream and local created using pkgbuilder instead of using static datasets. We should remove static datasets eventually.

Reply:
Yeah, I agree we should try to remove the static datasets eventually. But with this change, we at least use the exact same framework for the static datasets as with the pkgbuilder, it is just the source of the package data that changes.

Reply:
Correcting myself here. I now feel that having static datasets is better so that they serve both as examples and tests. And a non-coder can also read and write tests. Functions are tested like that. We should get there eventually.